### ISSUE TITLE
LogArchive always fails when using gfal2

### ISSUE THREAD

**Issue by bbockelm** [dmwm/WMCore#6978]
Hi,

`gfal2` requires that local filenames are prefixed with `file:/`; however, LogArchive tries to stage files in using the Unix path name (no `file:/`).  `lcg-cp` is forgiving on this and the transfer succeeds if you use `srmv2-lcg`.

FYI - there are currently 9 sites using `gfal2`.  That means we probably are missing quite a few log archives.

Note that OSG drops all support for `lcg-cp` next month and `LogCollect.py` hardcodes its use.  It should be expected that US sites will increasingly fail the LogArchive step.

Brian

**Comment by hufnagel** [comment]
Once #6895 is merged and deployed in the production agents this problems should disappear at least for all workflows using CMSS8X or later.

@amaltaro , you said the holdup on this was cmsweb. Since cmsweb was deployed yesterday, when can we deploy this patch to all the production agents ?

**Comment by bbockelm** [comment]
Agreed.  #6895 fixes this for `LogCollect` jobs.  Not sure if there are any other users of `StageInMgr` though?

**Comment by bbockelm** [comment]
@amaltaro - should we file a separate issue for the use of `srmv2-lcg` for log archive stageout?  When this is fixed, we'll construct a valid tarball and then fail stageout.

**Comment by hufnagel** [comment]
LogCollect stageout should really use xrdcp, there is no reason why we should use anything else to copy into EOS or Castor. With #6895 merged I can start development on that. Wanted to do this right away, but it would have been too much at once.

@bbockelm , you can open an issue on this if you want and @amaltaro can assign it to me.

**Comment by amaltaro** [comment]
The logCollect edmCopyUtils patch was pushed today to all the agents. However, since it works only for > 805 releases, don't we want to get it fixed for all the others? I believe there are quite a bit requests out there using older releases.

For the logcollect stage out, do we need to fix both stage out to castor and to EOS or only EOS? When I worked on the EOS code, I didn't manage to get it working with xrdcp ...

**Comment by bbockelm** [comment]
Either way - stageout via `srmv2-lcg` is not viable on OSG sites going forward.  Would it make sense to have a `xrdcp`-based plugin with `gfal2` as fallback?

For staging unmerged logs to local disk: @juztas said he would look at fixing the `gfal2` plugin tomorrow.

**Comment by hufnagel** [comment]
xrdcp works against both EOS and Castor. Done it from P5, which is an external site as far as CERN is concerned. So I doubt I wouldn't be able to make it work in a general way.

edmCopyUtil works for everything 800 or higher. Not sure if we need to provide fixes for older releases, that depends on how many pre-800 workflows we still run I guess.

**Comment by hufnagel** [comment]
@bbockelm , gfal2 fallback only makes sense if there is a possibility that some site won't have xrdcp installed but will have gfal2. I would say that chance is so remote that implementing a fallback isn't worth the time, but feel free to correct me if that is wrong.

**Comment by bbockelm** [comment]
Closing issue as xrootd-based stageout seems to be successfully deployed.
